samples/java-spring-cinema-booking-choreography/src/main/java/com/example/cinema/model/Show.java

```diff
@@ -0,0 +1,220 @@
+package com.example.cinema.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.vavr.Tuple2;
+import io.vavr.collection.HashMap;
+import io.vavr.collection.Map;
+import io.vavr.control.Either;
+import io.vavr.control.Option;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static com.example.cinema.model.CinemaApiModel.ShowCommandError.*;
+import static com.example.cinema.model.Show.SeatStatus.AVAILABLE;
+import static io.vavr.control.Either.left;
+import static io.vavr.control.Either.right;
+
+public record Show(String id, String title, Map<Integer, Seat> seats,
+                   Map<String, Integer> pendingReservations,
+                   Map<String, FinishedReservation> finishedReservations, int availableSeats) {
+
+    public static Show create(ShowEvent.ShowCreated showCreated) {
+        List<Tuple2<Integer, Seat>> seats = showCreated.seats().stream().map(seat -> new Tuple2<>(seat.number(), seat)).toList();
```

Comment:
A List of Tuple2 is the same as a Map. 
There must exist a map function to go from List to Map, so we don't need to introduce a type like Tuple2.

Reply:
I don't think so, but maybe they added sth in the mean time.